cmake/SetupBOUTThirdParty.cmake

```
@@ -167,11 +167,19 @@ if (BOUT_USE_NETCDF)
     # Use our own FindnetCDF module which uses nc-config
     find_package(netCDF REQUIRED)
     FetchContent_MakeAvailable(netcdf-cxx4)
+    target_link_libraries(bout++ PUBLIC netCDF::netcdf-cxx4)
   else()
-    find_package(netCDFCxx REQUIRED)
-    set(CONFIG_LDFLAGS "${CONFIG_LDFLAGS} ${netCDF_CXX_LIBRARY} ${netCDF_LIBRARIES}")
+    find_package(netCDFCxx)
+    if (netCDFCxx_FOUND)
+      set(CONFIG_LDFLAGS "${CONFIG_LDFLAGS} ${netCDF_CXX_LIBRARY} ${netCDF_LIBRARIES}")
+      target_link_libraries(bout++ PUBLIC netCDF::netcdf-cxx4)
+    else()
+      find_package(PkgConfig REQUIRED)
+      pkg_check_modules(NETCDF REQUIRED IMPORTED_TARGET netcdf-cxx4)
+      set(CONFIG_LDFLAGS "${CONFIG_LDFLAGS} ${netCDF_CXX_LIBRARY} ${NETCDF_LDFLAGS}")
+      target_link_libraries(bout++ PUBLIC ${NETCDF_LDFLAGS})
```

With `IMPORTED_TARGET` in `pkg_check_modules` we can do:

```suggestion
      target_link_libraries(bout++ PUBLIC PkgConfig::netcdf-cxx4)
```